pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -8511,3 +8511,78 @@ spec:
 func lessTaskResourceBindings(i, j v1beta1.TaskResourceBinding) bool {
 	return i.Name < j.Name
 }
+
+func TestReconcile_ApplyTaskLevelComputeResources(t *testing.T) {
+	multiStepsTask := parse.MustParseTask(t, `
+metadata:
+  name: foo-task
+  namespace: default
+spec:
+  steps:
+  - name: 1st-step
+    image: foo-image
+    command: 
+    - /foo-cmd
+  - name: 2nd-step
+    image: foo-image
+    command: 
+    - /foo-cmd
+`)
+	simplePipeline := parse.MustParsePipeline(t, `
+metadata:
+  name: foo-pipeline
+  namespace: default
+spec:
+  tasks:
+  - name: 1st-task
+    taskRef:
+      name: foo-task
+`)
+
+	for _, tc := range []struct {
+		name        string
+		pipeline    *v1beta1.Pipeline
+		pipelineRun *v1beta1.PipelineRun
+	}{{
+		name:     "apply-task-level-compute-resources",
+		pipeline: simplePipeline,
+		pipelineRun: parse.MustParsePipelineRun(t, `
+metadata:
+  name: foo-pipeline-run
+  namespace: default
+spec:
+  pipelineRef:
+    name: foo-pipeline
+  taskRunSpecs:
+  - pipelineTaskName: 1st-task
+    computeResources:
+      requests:
+        cpu: "500m"
+`)}} {
+		t.Run(tc.name, func(t *testing.T) {
+			d := test.Data{
+				PipelineRuns: []*v1beta1.PipelineRun{tc.pipelineRun},
+				Pipelines:    []*v1beta1.Pipeline{tc.pipeline},
+				Tasks:        []*v1beta1.Task{multiStepsTask},
+			}
+			prt := newPipelineRunTest(d, t)
+			defer prt.Cancel()
+
+			reconciledRun, clients := prt.reconcileRun("default", tc.pipelineRun.Name, []string{}, false)
+
+			if reconciledRun.Status.CompletionTime != nil {
+				t.Errorf("Expected a CompletionTime on invalid PipelineRun but was nil")
+			}
+
+			actual, err := clients.Pipeline.TektonV1beta1().TaskRuns("default").List(prt.TestAssets.Ctx, metav1.ListOptions{})
+
+			if err != nil {
+				t.Fatalf("Failure to list TaskRun's %s", err)
+			}
+			if len(actual.Items) != 1 {
+				t.Fatalf("Expected 1 TaskRuns got %d", len(actual.Items))
```

Comment:
This should test that the created taskrun has the expected spec

Reply:
yap. updated with verification logic.